sql-plugin/src/main/scala/com/nvidia/spark/rapids/PrioritySemaphore.scala

```diff
@@ -32,22 +31,30 @@ class PrioritySemaphore[T](val maxPermits: Int)(implicit ordering: Ordering[T])
   private val lock = new ReentrantLock()
   private var occupiedSlots: Int = 0
 
-  private case class ThreadInfo(priority: T, condition: Condition)
+  private case class ThreadInfo(priority: T, condition: Condition, numPermits: Int) {
+    var signaled: Boolean = false
+  }
+
+  private val threadComparator: Comparator[ThreadInfo] =
+    (o1: ThreadInfo, o2: ThreadInfo) => ordering.compare(o1.priority, o2.priority)
 
   // We expect a relatively small number of threads to be contending for this lock at any given
   // time, therefore we are not concerned with the insertion/removal time complexity.
-  private val waitingQueue: PriorityQueue[ThreadInfo] = PriorityQueue()(Ordering.by(_.priority))
+  private val waitingQueue: PriorityQueue[ThreadInfo] =
+    new PriorityQueue[ThreadInfo](threadComparator.reversed())
 
   def this()(implicit ordering: Ordering[T]) = this(PrioritySemaphore.DEFAULT_MAX_PERMITS)(ordering)
```

Comment:
Do we want this to be able to be default constructed?  Seems like we always want the caller to specify the number of permits.

Reply:
Yeah we don't need this